pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/AddTableCommand.java

```diff
@@ -102,33 +111,55 @@ public AddTableCommand setExecute(boolean exec) {
     return this;
   }
 
-  public boolean execute(JsonNode node)
-      throws IOException {
-    if (_controllerHost == null) {
-      _controllerHost = NetUtil.getHostAddress();
+  public void uploadSchema()
+      throws Exception {
+    File schemaFile;
+    Schema schema;
+    try {
+      schemaFile = new File(_schemaFile);
+      schema = Schema.fromFile(schemaFile);
+    } catch (Exception e) {
+      LOGGER.error("Got exception while reading Pinot schema from file: [" + _schemaFile + "]");
+      throw e;
     }
-    _controllerAddress = "http://" + _controllerHost + ":" + _controllerPort;
-
-    if (!_exec) {
-      LOGGER.warn("Dry Running Command: " + toString());
-      LOGGER.warn("Use the -exec option to actually execute the command.");
-      return true;
+    try (FileUploadDownloadClient fileUploadDownloadClient = new FileUploadDownloadClient()) {
+      fileUploadDownloadClient.addSchema(
+          FileUploadDownloadClient.getUploadSchemaHttpURI(_controllerHost, Integer.parseInt(_controllerPort)),
```

Comment:
Do we still want to upload schema if `exec` is not true?

Reply:
Dry run shouldn't upload schema. The check is at line 146